src/actions/wallet/L1/writeContractDeposit.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  type Abi,
+  type Account,
+  type Chain,
+  encodeFunctionData,
+  type EncodeFunctionDataParameters,
+  type Transport,
+  type WalletClient,
+  type WriteContractParameters,
+  type WriteContractReturnType,
+} from 'viem'
+import { OpStackL1Contract } from '../../../index.js'
+import type { GetL2Chain, L1ActionBaseType, ResolveChain } from '../../../types/l1Actions.js'
+import { writeSendMessage, type WriteSendMessageParameters } from './writeSendMessage.js'
+
+export type WriteContractDepositParameters<
+  TAbi extends Abi | readonly unknown[] = Abi,
+  TFunctionName extends string = string,
+  TChain extends Chain | undefined = Chain,
+  TAccount extends Account | undefined = Account | undefined,
+  TChainOverride extends Chain | undefined = Chain | undefined,
+> =
+  & { minGasLimit: number }
+  & L1ActionBaseType<GetL2Chain<ResolveChain<TChain, TChainOverride>>, typeof OpStackL1Contract.L1CrossDomainMessenger>
+  & WriteContractParameters<
+    TAbi,
+    TFunctionName,
+    TChain,
+    TAccount,
+    TChainOverride
+  >
+
+/**
+ * A L1 -> L2 version of Viem's writeContract. Can be used to call any L2 contract from L1.
+ * Internally uses writeSendMessage, which calls to the cross chain messenger contract.
+ * NOTE: msg.sender on the call to `address` will NOT be msg.sender on L1, but the L2CrossDomainMessenger.
+ *
+ * @param parameters - {@link WriteContractDepositParameters}
+ * @returns A [Transaction Hash](https://viem.sh/docs/glossary/terms.html#hash). {@link WriteContractReturnType}
+ */
+export async function writeContractDeposit<
```

Comment:
This works both l1 to l2 and l2 to l1 if you wanted to have a single hook that can go both directions

Reply:
sendMessage works on both, depositTransaction does not. But even then the behavior is very different depending on which direction you're going so I think having a single name might be more confusing than not.